lifecycle.go

```diff
@@ -200,3 +200,50 @@ func resolveAppendOptions(opts ...func(*AppendOptions)) *AppendOptions {
 	}
 	return defaults
 }
+
+// MigrationTarget describes the contract of the Migration lifecycle.
+//
+// This lifecycle mode is used to migrate C2SP tlog-tiles and static-ct
+// compliant logs into Tessera.
+type MigrationTarget interface {
+	// SetEntryBundle stores the provided serialised entry bundle at the location implied by the provided
+	// entry bundle index and partial size.
+	//
+	// Bundles may be set in any order (not just consecutively), and the implementation should integrate
+	// them into the local tree in the most efficient way possible.
```

Comment:
What happens if the same bundle is set twice?

1) Overwrite
2) Silently fail
3) Noisily fail
4) compare and fail if not equal

There are probably other strategies. This feels important to be on the interface to make all implementations do the same thing.

Reply:
Added language to the comment.